homeassistant/components/unifi/device_tracker.py

```diff
@@ -29,144 +34,122 @@
         self.controller = controller
-        self._detection_time = detection_time
-        self._ssid_filter = ssid_filter
-        self._monitored_conditions = monitored_conditions
-        self._clients = {}
 
     async def async_update(self):
-        """Get the clients from the device."""
-        try:
-            await self.controller.clients.update()
-            clients = self.controller.clients.values()
-
-        except aiounifi.LoginRequired:
-            try:
-                with async_timeout.timeout(5):
-                    await self.controller.login()
-            except (asyncio.TimeoutError, aiounifi.AiounifiException):
-                clients = []
-
-        except aiounifi.AiounifiException:
-            clients = []
-
-        # Filter clients to provided SSID list
-        if self._ssid_filter:
-            clients = [
-                client for client in clients
-                if client.essid in self._ssid_filter
-            ]
-
-        self._clients = {
-            client.raw['mac']: client.raw
-            for client in clients
-            if (dt_util.utcnow() - dt_util.utc_from_timestamp(float(
-                client.last_seen))) < self._detection_time
-        }
-
-    async def async_scan_devices(self):
-        """Scan for devices."""
-        await self.async_update()
-        return self._clients.keys()
-
-    def get_device_name(self, device):
-        """Return the name (if known) of the device.
-
-        If a name has been set in Unifi, then return that, else
-        return the hostname if it has been detected.
-        """
-        client = self._clients.get(device, {})
-        name = client.get('name') or client.get('hostname')
-        _LOGGER.debug("Device mac %s name %s", device, name)
-        return name
-
-    def get_extra_attributes(self, device):
-        """Return the extra attributes of the device."""
-        if not self._monitored_conditions:
-            return {}
-
-        client = self._clients.get(device, {})
+        """Synchronize state with controller."""
+        await self.controller.request_update()
+        await self.async_see()
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        return self.client.name or self.client.hostname
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this switch."""
+        return 'unifi-dt-{}'.format(self.client.mac)
+
+    @property
+    def available(self) -> bool:
+        """Return if controller is available."""
+        return self.controller.available
+
+    @property
+    def last_seen(self) -> dt_util.dt.datetime:
+        """Return the timestamp when device was last seen."""
+        return dt_util.utc_from_timestamp(float(self.client.last_seen))
+
+    @property
+    def device_state_attributes(self) -> dict:
+        """Return the device state attributes."""
         attributes = {}
-        for variable in self._monitored_conditions:
```

Comment:
Yes, I will revert to something similar behaviour as was before. As per our chat discussion earlier.

Reply:
How about I remove this option completely and then we can add back what is requested?

Reply:
I have a feeling that there won't be a standard subset of attributes that everyone will want. Some may want things like `rssi`, `tx_bytes`, `assoc_time`, etc for complex automations. Others will never want those as they'd simply be taking up recorder space. And since they're stored with the entities, you'd lose presence history if you disabled the recorder for them.

Since the additional attributes are only going to be used by those that have a good reason, I personally think they should be an opt-in list like today. It's a more complicated config, but only for those that want to complicate it.

Reply:
I'm keeping them for the moment